### PR TITLE
Gitlab fixes

### PR DESCRIPTION
**Pull request by bwrsandman**
- Try and find the most recent non-skipped build for status
- Avoid # in branch names

### CODE REVIEW

**Comment by yvaucher** [comment]
./runbot_gitlab/runbot_build.py:38:49: W291 trailing whitespace

Apart from this :+1:

File: runbot_gitlab/controllers/gitlab_ci_controller.py [hunk 1]
```diff
@@ -84,33 +85,40 @@ def builds(self, repo_id, sha, token=None):
...
+                elif state == 'testing':
+                    status = 'pending'
+                elif state == 'running':
+                    status = 'running'
+                elif result in ['ok', 'warn']:
+                    status = 'success'
+                else:
+                    status = 'unknown'
+            except IndexError:
+                logger.debug("No good builds found for commit %s is %s" % sha)
```
Comment (moylop260) [inline runbot_gitlab/controllers/gitlab_ci_controller.py]: ...%s is %s" % sha
Number of %s two vs params one

Comment (bwrsandman) [inline runbot_gitlab/controllers/gitlab_ci_controller.py]: Good catch

**Comment by max3903** [comment]
:+1:

**Pull request merged**